pkg/test/framework/suite.go

```diff
@@ -301,10 +307,40 @@ func (s *Suite) run() (errLevel int) {
 			}
 		}
 	}
+	s.writeOutput()
 
 	return
 }
 
+type SuiteOutcome struct {
+	Name         string
+	Environment  string
+	Multicluster bool
+	TestOutcomes []TestOutcome
+}
+
+func (s *Suite) writeOutput() {
+	// the ARTIFACTS env var is set by prow, and uploaded to GCS as part of the job artifact
+	artifactsPath := os.Getenv("ARTIFACTS")
+	if artifactsPath != "" {
+		ctx := rt.suiteContext()
+		out := SuiteOutcome{
+			Name:         ctx.Settings().TestID,
+			Environment:  ctx.Environment().EnvironmentName().String(),
+			Multicluster: ctx.Environment().IsMulticluster(),
+			TestOutcomes: ctx.testOutcomes,
+		}
+		outbytes, err := yaml.Marshal(out)
+		if err != nil {
+			log.Errorf("failed writing test suite outcome to yaml: %s", err)
+		}
+		err = ioutil.WriteFile(path.Join(artifactsPath, out.Name+".yaml"), outbytes, 0644)
```

Comment:
semi-related: i am pretty sure we can make the prow UI show these in a cool way like it does with JUnit

Reply:
with a lot of additional work, of course